app/core/audio/__init__.py

```diff
@@ -0,0 +1,18 @@
+import schema
+from core.audio.whisper import Whisper
+from custom_exceptions import GenericException
+
+
+async def handle_audio_data(
+    audio_data, 
+    transcription_type:schema.AudioTranscriptionType=schema.AudioTranscriptionType.WHISPER) -> str:
+    '''Takes audio input and returns the transcribed text as a string'''
+
+    if transcription_type == schema.AudioTranscriptionType.WHISPER:
+        whisper = Whisper()
+        user_message = whisper.transcribe_audio(audio_data)
+    
```

Comment:
In other sub modules we followed this pattern.
- __init__.py will contain the base/abstract/interface class , with common method signatures (`transcribe_audio()`, in this case)
- Other files in the same folder will have implementation of these. Now Whisper, others could come later.
- The implementation class would import the interface and inherit it. 

This `handle_audio_data()` might reside well in the conversation pipeline class, where we could have an optional audioTranscription component.

Reply:
OK great, thanks. I've tried to make those changes and just pushed them. Let me know if you have other thoughts or suggestions...